components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt

```diff
@@ -137,7 +137,6 @@ class DatabaseLoginsStorage(private val dbPath: String) : AutoCloseable, LoginsS
     @Throws(LoginsStorageException::class)
     override fun get(id: String): ServerPassword? {
         val json = nullableRustCallWithLock { raw, error ->
-            checkUnlocked()
```

Comment:
Oh, nice, so this is an example of where it's no longer a potential safety issue if we're locked and try to use an FFI call? 👍

Reply:
That call was actually redundant, nullableRustCallWithLock already checks this. That said, in general we can remove a lot of @Synchronized/synchronized(this) usage from the FFI now if we want, since rust uses a Mutex. I didnt do that here, since it would make this patch a lot harder to review.